chain/indexer/distributed/queue/tasks/index.go

```diff
@@ -28,8 +31,27 @@ type IndexTipSetPayload struct {
 	TraceCarrier *tracing.TraceCarrier `json:",omitempty"`
 }
 
+// Attributes returns a slice of attributes for populating tracing span attributes.
+func (i IndexTipSetPayload) Attributes() []attribute.KeyValue {
+	return []attribute.KeyValue{
+		attribute.Int64("height", int64(i.TipSet.Height())),
+		attribute.String("tipset", i.TipSet.Key().String()),
+		attribute.StringSlice("tasks", i.Tasks),
+	}
+}
+
+// MarshalLogObject implement ObjectMarshaler and allows user-defined types to efficiently add themselves to the
+// logging context, and to selectively omit information which shouldn't be
+// included in logs (e.g., passwords).
+func (i IndexTipSetPayload) MarshalLogObject(enc zapcore.ObjectEncoder) error {
+	enc.AddString("tipset", i.TipSet.Key().String())
+	enc.AddInt64("height", int64(i.TipSet.Height()))
+	enc.AddString("tasks", fmt.Sprint(i.Tasks))
+	return nil
+}
+
 // HasTraceCarrier returns true iff payload contains a trace.
-func (i *IndexTipSetPayload) HasTraceCarrier() bool {
+func (i IndexTipSetPayload) HasTraceCarrier() bool {
 	return !(i.TraceCarrier == nil)
```

Comment:
It's because we only ever use it as a value within methods. e.g. its instantiated as a value here: https://github.com/filecoin-project/lily/pull/981/files#diff-583b89249d2ef998d0f072521baf5ae75d506958a317c2ea3e6b6ad74ac7551eR75